lib/Yam/Agama/Pom/RebootTextmodePage.pm

```diff
@@ -22,4 +22,49 @@ sub reboot {
     enter_cmd 'reboot';
 }
 
+sub expect_is_shown {
+    my ($self, %args) = @_;
+    my $timeout = $args{timeout};
+
+    while (1) {
+        # agama log error
+        script_run('journalctl -u agama > /tmp/journal.log');
+        my $errors_in_log = script_output "grep 'ERROR -- :' /tmp/journal.log";
+        if ($errors_in_log) {
+            record_info('Found errors in agama journal',
+                "Error is $errors_in_log please check it",
+                result => 'fail');
+            die "agama failed with ($errors_in_log), please check the agama log.";
+        }
+        my $agama_auto_finished = script_run("grep 'agama-auto.service: Deactivated successfully' /tmp/journal.log");
+        if (!$agama_auto_finished) {
+            record_info('Install finished', "agama-auto Deactivated successfully", result => 'pass');
+            return;
+        }
+
+        # agama-auto service check
+        script_run('journalctl -u agama-auto > /tmp/journal_auto.log');
+        script_run('cat /tmp/journal_auto.log');
+
+        $errors_in_log = script_output "grep 'agama-auto service: Main process exited' /tmp/journal_auto.log | grep FAILURE";
+        if ($errors_in_log) {
+            record_info('Found errors in agama auto journal',
+                "Error is $errors_in_log please check it",
+                result => 'fail');
+            die "agama failed with ($errors_in_log), please check the agama log.";
+        }
+        # wait the installation to be finished.
+        my $yast_install_finished = script_run("grep 'Error output: Installation finished. No error reported.' /var/log/YaST2/y2log");
+        if ($yast_install_finished) {
+            record_info('Install finished', "$yast_install_finished", result => 'pass');
+            return;
+        }
+
+        die "timeout ($timeout) hit on during installation" if $timeout <= 0;
+        $timeout -= 30;
+        diag("left total timeout: $timeout");
+        next;
+    }
+}
+
```

Comment:
I don't understand why you need so much code here, I guess what you are trying to do is to to not wait so much time if the installation doesn't succeeds, that is good, not really required in this ticket, but what we could do is just to check every 30 seconds for the message of installation finished and if doesn't appear check for other message, it would be better if it would be just one single message and in the same file, do we really need to check also journal.log or agama-auto to do that? is not enough with y2log? In any case we should not try to display those errors or handle them, we would just die and the test should collect logs.

Reply:
This is a WIP, and I've asked in the [**agama chanel**](https://suse.slack.com/archives/C02TLF25571/p1730094821579579) for answer and still no reply. will be back on this once the bug was resolved. Thanks!

Reply:
It is not a bug, it is a feature (not a Jira one but something that developers plan to do), this ticket it is to workaround the problem while the feature is implemented:
https://trello.com/c/kdvo6Dtw/447-agama-add-an-agama-monitor-command

Reply:
OK, I'd rerun this branch to see if it works after [**their fix**](https://suse.slack.com/archives/C02TLF25571/p1729587009007949?thread_ts=1729582358.621149&cid=C02TLF25571) was in
>  fixed in https://github.com/agama-project/agama/pull/1688